### DISCUSSION TITLE
17.3

### DISCUSSION THREAD

**Discussion by bbakerman** [graphql-java/graphql-java#2554] [General]
This bug fix version of graphql-java provides new limits to help prevent Denial Of Service attacks induced by over parsing and validation.

Attackers can craft queries that consume lot of resources to parse and validate, which which ultimately invalid can deny real queries from being serviced.

https://github.com/graphql-java/graphql-java/pull/2549

https://github.com/graphql-java/graphql-java/pull/2553

There are new limits imposed by default.  Parsing will be terminated after 15000 tokens and only 100 validation errors will be captured.

We chose to put in defaults so that people will get some amount of bad query parse and validate DOS protection out of the box.

There are JVM wide methods to change the default on these if that's problematic for your implementation.

There is also a small fix in the ValueResolver

https://github.com/graphql-java/graphql-java/commit/8530366f24ba316075a63402473cb2a38ca36ab3

<hr /><em>This discussion was created from the release <a href='https://github.com/graphql-java/graphql-java/releases/tag/v17.2'>17.3</a>.</em>

**Comment by dfa1** [comment]
Hi @bbakerman!

very interesting release! We (at SIX) are planning to adopt this version as soon as possible. 

Just a minor typo: according to ParserOptions class the max number of tokens is

public static int MAX_QUERY_TOKENS = 15000;

whereas here is 1500?
